scripts/core/editor3/components/spellchecker/default-spellcheckers.tsx

```diff
@@ -33,29 +34,59 @@ function check(str: string): Promise<Array<ISpellcheckWarning>> {
 
             let lastOffset = 0;
 
-            str.split('\n').forEach((paragraph) => {
-                let matchArr;
-                let start;
-
-                // tslint:disable-next-line no-conditional-assignment
-                while ((matchArr = regex.exec(paragraph)) !== null) {
-                    start = matchArr.index;
-                    if (!spellcheck.isCorrectWord(matchArr[0])) {
-                        info.push({
-                            startOffset: lastOffset + start,
-                            text: matchArr[0],
-                            suggestions: null,
-                        });
-                    }
-                }
+            let matchArr;
+            let start;
 
-                lastOffset += paragraph.length;
-            });
+            // tslint:disable-next-line no-conditional-assignment
+            while ((matchArr = regex.exec(str)) !== null) {
+                start = matchArr.index;
+                if (!spellcheck.isCorrectWord(matchArr[0])) {
+                    info.push({
+                        startOffset: lastOffset + start,
+                        text: matchArr[0],
+                        suggestions: null,
+                        type: 'spelling',
+                    });
+                }
+            }
 
             return info;
         });
 }
 
 export function getSpellchecker(language: string): ISpellchecker {
-    return {check, getSuggestions, actions};
+    const spellcheckerName = ({
+        fr: 'grammalecte',
+        nl: 'leuven_dutch',
+    })[language];
+
+    if (spellcheckerName != null) {
+        return {
+            check: (str: string) => new Promise((resolve, reject) => {
+                httpRequestJsonLocal<{errors: Array<ISpellcheckWarning>}>({
+                    method: 'POST',
+                    payload: {spellchecker: spellcheckerName, text: str},
+                    path: '/spellchecker',
+                    onSuccess: (json) => {
+                        resolve(json.errors);
+                    },
+                    onError: () => reject(),
```

Comment:
I think `reject` should include the error so we can `.catch(err)`. This should be enought:

```js
onSuccess: ...,
onError: reject,
```

Reply:
I already log the error in `function httpRequest(options: IHttpRequestOptions)`. If I were to pass an argument to `reject` I should have a use for it, which I don't. Even if we were to display a message to a user, the only relevant thing the message could say is that there was a network error, which is helpful but would be better off handled at the application level, not at feature/component level. I've created `SDESK-4464` to address that.